pyathena/sqlalchemy/base.py

```diff
@@ -789,7 +789,26 @@ def _prepared_columns(
                         or column.name in conn_partitions
                         or f"{table.name}.{column.name}" in conn_partitions
                     ):
-                        partitions.append(f"\t{processed}")
+                        #https://docs.aws.amazon.com/athena/latest/ug/querying-iceberg-creating-tables.html#querying-iceberg-partitioning
+                        if is_iceberg:
+                            partition_transform = column_dialect_opts["partition_transform"]
+                            if partition_transform:
+                                if AthenaPartitionTransform.is_valid(partition_transform):
+                                    if partition_transform == AthenaPartitionTransform.PARTITION_TRANSFORM_BUCKET:
+                                        bucket_count = column_dialect_opts["partition_transform_bucket_count"]
+                                        if bucket_count:
+                                            partitions.append(f"\t{partition_transform}({bucket_count},{self.preparer.format_column(column)})")
+                                    elif partition_transform == AthenaPartitionTransform.PARTITION_TRANSFORM_TRUNCATE:
+                                        truncate_length = column_dialect_opts["partition_transform_truncate_length"]
+                                        if truncate_length:
+                                            partitions.append(f"\t{partition_transform}({truncate_length},{self.preparer.format_column(column)})")
+                                    else:
+                                        partitions.append(f"\t{partition_transform}({self.preparer.format_column(column)})")
+                            else:
+                                partitions.append(f"\t{self.preparer.format_column(column)}")
+                            columns.append(f"\t{processed}")
+                        else:
+                            partitions.append(f"\t{processed}")
                     else:
                         columns.append(f"\t{processed}")
```

Comment:
Since the if statements are deeply nested, it may be a good idea to split them into multiple methods and extract them.

Reply:
> Since the if statements are deeply nested, it may be a good idea to split them into multiple methods and extract them.

I'll refactor this code once the tests pass.